### PR TITLE
QA-387: add utf8-collection tests

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

Implement UTF-8 support checks

- [x] :pizza: New feature

### Checklist

- [ ] sidelining PRs
  - [ ] [ArangoDB Pullrequest](https://github.com/arangodb/arangodb/pull/)
  - [ ] [RTA Pullrequest](https://github.com/arangodb/release-test-automation/pull/350)

### CODE REVIEW

File: test_data/makedata_suites/021_foxx_utf-8.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      } catch (e) {
+        print(route + " Caught - need to retry. " + JSON.stringify(e));
+      }
+      internal.sleep(3);
+    }
+    throw new Error("foxx route '" + route + "' not ready on time!");
+  };    
+  return {
+    isSupported: function (version, oldVersion, options, enterprise, cluster) {
+      return options.testFoxx;
```
Comment (alexbakharew) [inline test_data/makedata_suites/021_foxx_utf-8.js:51]: Should we rely here on version and oldVersion?

File: test_data/makedata_suites/021_foxx_utf-8.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    },
+    clearDataDB: function (options, isCluster, isEnterprise, database, dbCount) {
+      // All items created must contain dbCount
+      let dbName = `${extendedNames[0]}FoxxTest${extendedNames[3]}_${dbCount}`;
+      db._useDatabase(dbName);
+      print(`deleting foxx ${dbCount}${dbName}`);
+      print("uninstalling Itzpapalotl");
+      deleteFoxx(`/itz_${dbCount}`);
+
+      print("uninstalling crud");
```
Comment (alexbakharew) [inline test_data/makedata_suites/021_foxx_utf-8.js:143]: Do we really need this output?

File: test_data/makedata_suites/021_foxx_utf-8.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      internal.sleep(3);
+    }
+    throw new Error("foxx route '" + route + "' not ready on time!");
+  };    
+  return {
+    isSupported: function (version, oldVersion, options, enterprise, cluster) {
+      return options.testFoxx;
+    },
+    makeDataDB: function (options, isCluster, isEnterprise, database, dbCount) {
+      let dbName = `${extendedNames[0]}FoxxTest${extendedNames[3]}_${dbCount}`;
```
Comment (alexbakharew) [inline test_data/makedata_suites/021_foxx_utf-8.js:54]: Why we use only 2 items from extendedNames array?

File: test_data/makedata_suites/102_collection_utf8.js [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+      baseName = `M${baseName}_${dbCount}_${extendedNames[3]}`;
+      print(`102: creating ${baseName}`);
+      db._createDatabase(baseName);
+    },
+    makeData: function (options, isCluster, isEnterprise, dbCount, loopCount) {
+      // All items created must contain dbCount and loopCount
+      // Create a few collections:
+      print(`102: using ${baseName}`);
+      db._useDatabase(baseName);
+      let c = createCollectionSafe(`c_${loopCount}${extendedNames[0]}`, 3, 2);
```
Comment (alexbakharew) [inline test_data/makedata_suites/102_collection_utf8.js:27]: dbCount is missing here?

File: test_data/makedata_suites/102_collection_utf8.js [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+      progress('createIndex5');
+      createIndexSafe({col: cmulti, type: "hash", fields: ["a"], unique: false, name: extendedNames[6]});
+      progress('createIndex6');
+      createIndexSafe({col: cmulti, type: "skiplist", fields: ["b", "c"], name: extendedNames[7]});
+      progress('createIndex7');
+      createIndexSafe({col: cmulti, type: "geo", fields: ["position"], geoJson: true, name: extendedNames[8]});
+      progress('createIndexGeo8');
+      createIndexSafe({col: cmulti, type: "fulltext", fields: ["text"], minLength: 6, name: extendedNames[0]});
+      progress('createIndexFulltext9');
+
```
Comment (alexbakharew) [inline test_data/makedata_suites/102_collection_utf8.js:63]: Please add inverted index here as well

File: test_data/makedata_suites/102_collection_utf8.js [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+      if (db._query(aql`FOR x IN ${cempty} RETURN x`).toArray().length !== 0) { throw new Error("Grapefruit"); }
+      progress("102: query 5");
+      if (db._query(aql`FOR x IN ${cgeo} FILTER x.a == "id20473" RETURN x`).toArray().length !== 1) { throw new Error("Bean"); }
+      progress("102: query 6");
+      if (db._query(aql`FOR x IN ${cunique} FILTER x.a == "id32236" RETURN x`).toArray().length !== 1) { throw new Error("Watermelon"); }
+      progress("102: query 6");
+      if (db._query(aql`FOR x IN ${cmulti} FILTER x.a == "id32847" RETURN x`).toArray().length !== 1) { throw new Error("Honeymelon"); }
+      progress("102: queries done");
+      db._useDatabase('_system');
+    },
```
Comment (alexbakharew) [inline test_data/makedata_suites/102_collection_utf8.js:247]: Please add query for inverted index here as well

File: test_data/makedata_suites/102_collection_utf8.js [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+      try {
+        db._drop(`c_${loopCount}${extendedNames[0]}`);
+      } catch (e) {}
+      progress();
+      try {
+        db._drop(`chash_${loopCount}${extendedNames[1]}`);
+      } catch (e) {}
+      progress();
+      try {
+        db._drop(`cskip_${loopCount}${extendedNames[2]}`);
```
Comment (alexbakharew) [inline test_data/makedata_suites/102_collection_utf8.js:274]: leftovers should be cleaned, I think

File: test_data/makedata_suites/402_views.js [hunk 8]
```diff
@@ -812,7 +812,7 @@ function deleteAnalyzer_400(testgroup, analyzerName){
     let headers = {};
     headers['accept'] = 'application/json';
     headers["Authorization"] = `Bearer ${jwt_key}`;
-    let clusterHealth = undefined;
+    let clusterHealth;
     let count = 0;
```
Comment (alexbakharew) [inline test_data/makedata_suites/402_views.js:816]: Count should be 5?

File: test_data/makedata_suites/403_utf8-name-views.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        // includeAllFields: true
+      };
+      view1.properties(meta);
+
+      cview1.insert([
+        {"animal": "cat", "name": "tom"},
+        {"animal": "mouse", "name": "jerry"},
+        {"animal": "dog", "name": "harry"}
+      ]);
+      progress('403: createView3');
```
Comment (alexbakharew) [inline test_data/makedata_suites/403_utf8-name-views.js:39]: We don't have createView3

File: test_data/makedata_suites/403_utf8-name-views.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      progress('403: createView1');
+      let viewName1 = `old_view1_403_${loopCount}${extendedNames[6]}`;
+      let view1 = createSafe(viewName1,
+                             viewname => {
+                               return db._createView(viewname, "arangosearch", {});
+                             }, viewname => {
+                               return db._view(viewname);
+                             }
+                            );
+      progress('createView2');
```
Comment (alexbakharew) [inline test_data/makedata_suites/403_utf8-name-views.js:25]: We don't have createView2

File: test_data/makedata_suites/403_utf8-name-views.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+      return semver.gte(currentVersionSemver, "3.11.0") && semver.gte(oldVersionSemver, "3.11.0");
+    },
+    makeData: function (options, isCluster, isEnterprise, dbCount, loopCount) {
+      // All items created must contain dbCount and loopCount
+      print(`403: making data ${dbCount} ${loopCount}`);
+      let viewCollectionName = `old_cview1_${loopCount}${extendedNames[3]}`;
+      let cview1 = createCollectionSafe(viewCollectionName, 3, 1);
+      progress('403: createView1');
+      let viewName1 = `old_view1_403_${loopCount}${extendedNames[6]}`;
+      let view1 = createSafe(viewName1,
```
Comment (alexbakharew) [inline test_data/makedata_suites/403_utf8-name-views.js:18]: Along with `arangosearch` view, we should test `search-alias` as well

**Review by alexbakharew — CHANGES_REQUESTED** [review]
Need to add inverted index and search-alias functionality

File: test_data/makedata_suites/021_foxx_utf-8.js (hunk 2 above)
Comment (dothebart) [reply test_data/makedata_suites/021_foxx_utf-8.js:143]: rta-makedata needs much output to track potential lockups.

File: test_data/makedata_suites/021_foxx_utf-8.js (hunk 3 above)
Comment (dothebart) [reply test_data/makedata_suites/021_foxx_utf-8.js:54]: to have a rough overview.

File: test_data/makedata_suites/102_collection_utf8.js (hunk 4 above)
Comment (dothebart) [reply test_data/makedata_suites/102_collection_utf8.js:27]: hm, 100 doesn't have these either...

File: test_data/makedata_suites/102_collection_utf8.js (hunk 5 above)
Comment (dothebart) [reply test_data/makedata_suites/102_collection_utf8.js:63]: created subsequent QA-456 task for this.

Comment (alexbakharew) [reply test_data/makedata_suites/102_collection_utf8.js:63]: @dothebart I think you can copy definitions from 101 and 402 test suites. It doesn't look like a huge work to add missing stuff

File: test_data/makedata_suites/403_utf8-name-views.js (hunk 11 above)
Comment (dothebart) [reply test_data/makedata_suites/403_utf8-name-views.js:18]: QA-457 added

File: test_data/makedata_suites/403_utf8-name-views.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        {"animal": "mouse", "name": "jerry"},
+        {"animal": "dog", "name": "harry"}
+      ]);
+      progress('403: createView3');
+    },
+    checkData: function (options, isCluster, isEnterprise, dbCount, loopCount, readOnly) {
+      print(`403: checking data ${dbCount} ${loopCount}`);
+      // Check view:
+      let view1 = db._view(`old_view1_403_${loopCount}${extendedNames[6]}`);
+      if (!view1.properties().links.hasOwnProperty(`old_cview1_${loopCount}${extendedNames[3]}`)) {
```
[the remark this answers is not in the document]
Comment (dothebart) [reply test_data/makedata_suites/403_utf8-name-views.js:45]: QA-435

**Pull request merged**